File: Python/pass_generator.py
#Generator hesel
import time
import random
#for i in range (0,100):


letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z']
numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
special_char = ['%', '#', '$', '!', '&', '(', ')', '*', '+', '?']

letters_len = int(input("Kolik znaku?\n"))
numbers_len = int(input("Kolik cisel?\n"))
special_char_len = int(input("Kolik specialnich znaku?\n"))
password = []
start_time = time.time()
for i in range(0, letters_len):
    password.append(letters[random.randint(0, len(letters)-1)])
        
for i in range(0, numbers_len):
    password.append(numbers[random.randint(0, len(numbers)-1)])
        
for i in range(0, special_char_len):
    password.append(special_char[random.randint(0, len(special_char)-1)])
  
    
    #Prehazeni. Normalne existuje random.shuffle()
random.shuffle(password)
    #Prevod listu na str. Noralne existuje "".join()
password = "".join(password)
        
print(password)
    
    

end_time = time.time()
elapsed_time = end_time - start_time

print(f"Elapsed time:{elapsed_time}")

   

File: C/pass_generator.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
     
char generateRandomChar() {


    // Define the character sets
    const char letters[] = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
    const char numbers[] = "0123456789";
    const char specialChars[] = "!@#$%^&*()_+-=[]{}|;:'\",.<>/?";

    int choice = rand() % 3;  // 0 for letters, 1 for numbers, 2 for special characters

    if (choice == 0) {
        return letters[rand() % 52];
    } else if (choice == 1) {
        return numbers[rand() % 10];
    } else {
        return specialChars[rand() % 29];
    }
}

void generatePassword(int numLetters, int numNumbers, int numSpecialChars) {

    printf("Generated Password: ");
    for (int i = 0; i < numLetters; i++) {
        printf("%c", generateRandomChar());
    }
    for (int i = 0; i < numNumbers; i++) {
        printf("%c", generateRandomChar());
    }
    for (int i = 0; i < numSpecialChars; i++) {
        printf("%c", generateRandomChar());
    }
    printf("\n");
}

int main() {
    srand(time(0)); // Seed the random number generator with the current time
    clock_t start_time = clock();
    int numLetters, numNumbers, numSpecialChars;

    printf("Enter the number of letters in the password: ");
    scanf("%d", &numLetters);
    printf("Enter the number of numbers in the password: ");
    scanf("%d", &numNumbers);
    printf("Enter the number of special characters in the password: ");
    scanf("%d", &numSpecialChars);

   
    generatePassword(numLetters, numNumbers, numSpecialChars);
    clock_t end_time = clock();

    double elapsed_time = (double)(end_time - start_time) / CLOCKS_PER_SEC;
    printf("Elapsed Time: %.5f seconds\n", elapsed_time);

    return 0;
}
